internal/dcache/metadata_manager/metadata_manager_impl.go

```diff
@@ -418,13 +432,64 @@ func (m *BlobMetadataManager) getFile(filePath string) ([]byte, int64, error) {
 
 	//if fileSize < 0 {
 	if fileSize < -1 {
-		log.Warn("GetFile:: Size is negative for path %s: %d", path, fileSize)
-		common.Assert(false, "size cannot be negative", fileSize)
-		return nil, -1, fmt.Errorf("size is negative for path %s: %d", path, fileSize)
+		err := fmt.Errorf("Size is negative for path %s: %d", path, fileSize)
+		log.Warn("GetFile:: %v", err)
+		common.Assert(false, err)
+		return nil, -1, "", err
 	}
 
 	log.Debug("GetFile:: Size for path %s: %d", path, fileSize)
-	return data, fileSize, nil
+
+	// Extract the state from the blob metadata prop.
+	state, ok := prop.Metadata["state"]
+	if !ok {
+		err := fmt.Errorf("GetFile:: File state  not found in metadata for path %s", path)
+		log.Err("%v", err)
+		common.Assert(false, err)
+		return nil, -1, "", err
+	}
+
+	var fileState dcache.FileState
+	if *state == string(dcache.Deleting) || *state == string(dcache.Ready) || *state == string(dcache.Writing) {
+		fileState = dcache.FileState(*state)
+	} else {
+		err := fmt.Errorf("GetFile:: Invalid File state: [%s] found in metadata for path: %s", *state, path)
+		log.Err("%v", err)
+		common.Assert(false, err)
+		return nil, -1, fileState, err
+	}
+
+	return data, fileSize, fileState, nil
+}
+
+func (m *BlobMetadataManager) updateFileStateToDeleting(filePath string, fileMetadata []byte, fileSize int64) error {
+	path := filepath.Join(m.mdRoot, "Objects", filePath)
+	log.Debug("DeleteFile:: Updated State of the metadata blob to deleting path: %s in storage", path)
+	common.Assert(len(filePath) > 0)
+
+	openCount := "0"
+	sizeStr := strconv.FormatInt(fileSize, 10)
+	state := string(dcache.Deleting)
+	metadata := map[string]*string{
+		"opencount":           &openCount,
+		"cache_object_length": &sizeStr,
+		"state":               &state,
+	}
+
+	err := m.storageCallback.PutBlobInStorage(internal.WriteFromBufferOptions{
+		Name:                   path,
+		Data:                   fileMetadata,
+		Metadata:               metadata,
+		IsNoneMatchEtagEnabled: false,
+		EtagMatchConditions:    "",
+	})
```

Comment:
we should only set the state property to deleting and nothing else - none of the other properties and also not file data. So you may want to call m.storageCallback.SetMetaPropertiesInStorage(). It should get the properties and the only update the state to deleting.
Just like updateHandleCount() updates only the opencount property.